TappyBle/TappyBleDevice.swift

```diff
@@ -25,7 +25,7 @@ import Foundation
 
 @objc
 public class TappyBleDevice : NSObject {
-    var deviceName : String
+    @objc public var deviceName : String
```

Comment:
I'm uncertain as to why this is needed.  What error is the name() method giving you?

Reply:
Here's the error:

JS ERROR TypeError: tappyDevice.name is not a function. (In 'tappyDevice.name()', 'tappyDevice.name' is undefined)

Reply:
I'm using it like this: 
let name = tappyDevice.name() 

where tappyDevice is a TappyBleDevice returned by the tappyFoundListener. The deviceId is also coming as undefined when I print out tappyDevice.deviceId

Reply:
Using the .name() method has no issues on my end.  I think it may be that you're using this in an odd way?  I'm not sure what you mean by " tappyDevice is a TappyBleDevice returned by the tappyFoundListener"  That's not quite how it is intended to work.  Your function that you set as a "Tappy found listener" should receive this TappyBleDevice object when it is called.  If you can't call .name() or access the .deviceId property perhaps something else is wrong.   Perhaps it's a namespace conflict?  Send me you full JS code perhaps I'll notice something

Reply:
The whole repo is here: https://github.com/fhackenb/nativescript-tappy The relevant code is under src/tappy.ios.ts in the constructor

Reply:
this.scanner.setTappyFoundListenerWithListener( (tappyDevice: TappyBleDevice) => {
            console.log("Found tappy:");
            let name = tappyDevice.name(); // this line is throwing that error
            let deviceId = tappyDevice.deviceId;
            console.log("Name:", name);
            console.log("Device id:", tappyDevice.deviceId);
            this.device = tappyDevice;
            const tappyFoundEvent = {
                eventName: "tappyFound",
                object: this,
                device: this.device
            };
            this.notify(tappyFoundEvent);
        });

Reply:
Yea, I realize I explained that badly. My listener is a function and is receiving the TappyBleDevice when it's invoked

Reply:
I am unfamiliar with typescript, but my suspicion is that this is getting invoked with an uninitialized TappyBleDevice object. Perhaps it's not getting called only when you discover a TappyBLE reader?  Your code above is slightly different than what's currently posted. I'd be interested to know  what the code `console.log(JSON.stringify(tappyDevice));` produces

Reply:
When I do that it outputs an empty object: {}. It does seem to be invoked correctly though - it is consistently only triggered after I turn on the tappy (while it's scanning)

Reply:
Then it's definitely being invoked with an empty TappyBleDevice object then.  Try having the TappyBLE already on before you start the scan, does that make a difference?

Reply:
Nope, having it on to begin with is still throwing the empty object. The device works with your demo app (though I haven't output the TappyBleDevice object) so it's definitely not a hardware problem

Reply:
Could it be that whatever method is triggering the TappyFoundListener isn't fully exported to @objc?

Reply:
Not sure.  I have incorporated some of your changes into the latest version in the master.  I will hold off on publishing the updated cocoapod though until we look into this more.  

I recommend that you perhaps add annotations to line 30 of TappyBleScanner.swift, where the TappyFoundListener is declared.  If that works for you, then I will incorporate that and release a new Cocoapod.

Reply:
Ok, I added an annotation there and made it public. Also added one to the statusListener in the line below it

Reply:
Okay, and when you test it in your app, does it solve the problem?

Reply:
When I test it in my app, it has the same result (empty object). I can see the name is being console logged in centralManager right before it's passed to the tappyFoundListener, so I'm gonna try a couple other things to debug / see if I can get it to work

Reply:
I've tried a few things ( mostly adding annotations in different places) and haven't been able to resolve this yet. The name always outputs prior to being passed to tappyFoundListener but the object always is empty when the listener gets a hold of it. 

One thing I haven't been able to figure out (probably a Swift syntax thing) is how some listener functions have a prefix of sorts before the listener i.e:

@objc public func setStatusListener(statusReceived listener: @escaping (TappyBleScannerStatus) -> ()) 

or

setDataListener(receivedBytes listener: @escaping ([UInt8]) -> ())

What is the purpose of the "statusReceived" and "receivedBytes" text before the listener? I've tried adding a similar line in the setTappyFoundListener definition but that just throws a runtime error. Could this be something to look into? May be a shot in the dark, but it's the only real difference I noticed between the setStatusListener which works properly in my plugin and the tappyFoundListener which doesn't

Reply:
Another tidbit: I tried connecting to the tappy even though the name and deviceId are showing up as undefined. Before doing connect the status is STATUS_CLOSED (gotten by calling getLatestStatus()). Then calling connect() that status is changed to STATUS_CONNECTING and then to STATUS_READY indicating that it is working anyway. Tomorrow I'm gonna attempt to see if read / write operations are working even though the deviceId and name appear to be undefined

Reply:
The prefix is just an argument label in Swift, if you think those are causing issues by all means remove them and see if that changes things.   If it does not, and you still get the empty TappyBleDevice object getting passed to the listener, then it could be something else that we don't realize.  I can't speak for Typescript or Nativescript plugin development I'm afraid as I'm unfamiliar with those technologies.

Reply:
I played around with removing those argument labels a little bit and it doesn't seem to do anything. I'm going to attempt to proceed with developing the plugin as getting the name isn't essential to me right now - though I will probably revisit it later. In the meantime could you make the CocoaPod from the pull request? I still did need to add a few of those annotations for my app

Reply:
I'm going to update the pod now.  The current master has your annotations incorporated already, so I will close this pull request after that. The pod will be at v0.1.8